python/ray/_raylet.pyx

```diff
@@ -1284,15 +1284,14 @@ cdef void execute_task(
                 actor_title = f"{class_name}({args!r}, {kwargs!r})"
                 core_worker.set_actor_title(actor_title.encode("utf-8"))
 
-            # Record the task id via magic token in the log file.
-            # This will be used to locate the beginning of logs from a task.
-            attempt_number = core_worker.get_current_task_attempt_number()
-            task_attempt_magic_token = "{}{}-{}\n".format(
-                ray_constants.LOG_PREFIX_TASK_ATTEMPT_START, task_id.hex(),
-                attempt_number)
-            # Print on both .out and .err
-            print(task_attempt_magic_token, end="")
-            print(task_attempt_magic_token, file=sys.stderr, end="")
+            # Record the log file offsets instead
+            if core_worker.current_actor_max_concurrency() == 1:
+                # If it's not a concurrent actor task, we will record the start
+                # offset of the log file so that we could know where the task log
+                # begins.
+                # We are skipping concurrent actor tasks because high contention
+                # and slow IO on concurrent actors would result in perf regression.
+                worker.record_task_log_start()
```

Comment:
Make core_worker.current_actor_max_concurrency() == 1: part of start and end API?